x-pack/plugins/ingest_pipelines/public/application/components/pipeline_form/pipeline_form.tsx

```diff
@@ -264,28 +290,49 @@ export const PipelineForm: React.FunctionComponent<Props> = ({
         {/* Form submission */}
         <EuiFlexGroup justifyContent="spaceBetween">
           <EuiFlexItem grow={false}>
-            <EuiFlexGroup>
-              <EuiFlexItem grow={false}>
-                <EuiButton
-                  fill
-                  color="secondary"
-                  iconType="check"
-                  onClick={form.submit}
-                  data-test-subj="submitButton"
-                  disabled={form.isSubmitted && form.isValid === false}
-                  isLoading={isSaving}
-                >
-                  {
-                    <FormattedMessage
-                      id="xpack.ingestPipelines.form.createButtonLabel"
-                      defaultMessage="Create pipeline"
-                    />
-                  }
-                </EuiButton>
-              </EuiFlexItem>
-            </EuiFlexGroup>
+            <EuiButton
+              fill
+              color="secondary"
+              iconType="check"
+              onClick={form.submit}
+              data-test-subj="submitButton"
+              disabled={form.isSubmitted && form.isValid === false}
+              isLoading={isSaving}
+            >
+              {isSaving ? (
+                <FormattedMessage
+                  id="xpack.ingestPipelines.form.savingButtonLabel"
+                  defaultMessage="Saving..."
+                />
+              ) : (
+                saveButtonLabel
+              )}
+            </EuiButton>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiButtonEmpty
+              onClick={() => setIsRequestVisible(prevIsRequestVisible => !prevIsRequestVisible)}
+            >
+              {isRequestVisible ? (
+                <FormattedMessage
+                  id="xpack.ingestPipelines.form.hideRequestButtonLabel"
+                  defaultMessage="Hide request"
+                />
+              ) : (
+                <FormattedMessage
+                  id="xpack.ingestPipelines.form.showRequestButtonLabel"
+                  defaultMessage="Show request"
+                />
+              )}
+            </EuiButtonEmpty>
           </EuiFlexItem>
         </EuiFlexGroup>
+        {isRequestVisible ? (
+          <PipelineRequestFlyout
+            pipeline={form.getFormData() as Pipeline}
```

Comment:
@sebelga Is this the best way to get the form data for the flyout? The downside being that it doesn’t update if the user makes changes to the form with the flyout open. I considered using the `<FormDataProvider>`, but I see it returns the raw data format.

Reply:
This is one way, but as you realized it does not update when fields change. This is on purpose as the fields serializers and form serializer might be expensive.

In this case, it seems that it is OK, here is how to do it

Create a separate component to connect to the form and subscribe to its changes. It is currently required to have a separate component (instead of putting the subscribing logic inside the current `<PipelineForm />`) because the `form` object is not memoized in the lib. This is something I will change in a future update.

So for now, create a new component 

```js
import React, { useState, useEffect } from 'react';

import { Pipeline } from '../../../common/types';
import { useFormContext } from '../../shared_imports';
import { PipelineRequestFlyout } from './pipeline_request_flyout';

export const PipelineRequestFlyoutProvider = ({ closeFlyout }: { closeFlyout: () => void }) => {
  const form = useFormContext();
  const [formData, setFormData] = useState<Pipeline>({} as Pipeline);

  useEffect(() => {
    const subscription = form.subscribe(async ({ isValid, validate, data }) => {
      const isFormValid = isValid ?? (await validate());
      if (isFormValid) {
        setFormData(data.format() as Pipeline);
      }
    });

    return subscription.unsubscribe;
  }, [form]);

  return <PipelineRequestFlyout pipeline={formData} closeFlyout={closeFlyout} />;
};
```

Replace the export with the provider

```js
// index.ts
export { PipelineRequestFlyoutProvider as PipelineRequestFlyout } from './pipeline_request_flyout_provider';
```

And change the JSX in `PipelineForm`

```js
// pipeline_form.tsx

{isRequestVisible ? (
  <PipelineRequestFlyout
    closeFlyout={() => setIsRequestVisible(prevIsRequestVisible => !prevIsRequestVisible)}
  />
) : null}
```

Reply:
I also realized there is a bug in the `<EuiCodeBlock>` component. The value copied to the clipboard does not update when the content change.

I added a fix on my PR to handle that. I will ask the @elastic/eui-design if this is a known issue.

Reply:
Thanks @sebelga for the detailed explanation and PR! I tested your changes and merged into this branch.